frontend/src/modules/studySamples/services.ts

```diff
@@ -0,0 +1,155 @@
+import { FMSSampleNextStepByStudy, FMSId, FMSStepHistory } from "../../models/fms_api_models"
+import { createItemsByID, Study, Workflow } from "../../models/frontend_models"
+import store from "../../store"
+import api from "../../utils/api"
+import { fetchLibrariesForSamples, fetchProcesses, fetchProcessMeasurements, fetchSamples, fetchStudies, fetchUsers, fetchWorkflows } from "./cache"
+import { CompletedStudySample, StudySampleList, StudySampleStep } from "./models"
+
+enum StudySamplesErrorCode {
+	DEPENDENCY_NOT_FOUND,
+	DEPENDENCY_NOT_READY,
+	FETCH_ERROR
+}
+
+export class StudySamplesError extends Error {
+
+	constructor(public readonly code: StudySamplesErrorCode, message: string) {
+		super(message)
+	}
+}
+
+export async function loadStudySamples(studyID: FMSId) {
+	
+	const study = (await fetchStudies([studyID])).find(obj => obj.id === studyID)
+	if(! study) {
+		throw new StudySamplesError(StudySamplesErrorCode.DEPENDENCY_NOT_FOUND, `Study "${studyID}" not found.`)
```

Comment:
I guess defining a new exception class for these errors is a bit of overkill. I've replaced them with generic Errors, and got rid of the codes. You're right that the errors are too fine-grained to be useful for the user. All that really matters is that a helpful error message gets logged to help us diagnose any problems the users run into in production. I'll still check for things like the study being undefined, since if I don't then an unhelpful "cannot read id of undefined" error will be thrown at some random point later, making it much harder to diagnose the issue.